From 217169f68a3791a2c7f113bf488e20a59f40b465 Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Tue, 18 Jul 2023 21:36:00 +0800
Subject: [PATCH 12/54] Force kernelflinger to enter into fastboot mode

Build kernelfliner as fb4sbl.elf file to support fastboot features

Tracked-On: OAM-111239
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
---
 kernelflinger.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/kernelflinger.c b/kernelflinger.c
index 5e57c57..001ebed 100644
--- a/kernelflinger.c
+++ b/kernelflinger.c
@@ -1267,6 +1267,14 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table)
 		return ret;
 	}
 
+	/* The code is only availble for fb4sbl.elf image which is used
+	 * as ELK file in non efi boot. It will force bootloader enter
+	 * into fastboot mode.
+	 */
+#ifdef __FORCE_FASTBOOT
+	enter_fastboot_mode(boot_state);
+#endif
+
 	/* No UX prompts before this point, do not want to interfere
 	 * with magic key detection
 	 */
-- 
2.25.1

